pkg/aws/permissions.go

```diff
@@ -36,11 +36,11 @@ type SimulateParams struct {
 
 // checkPermissionsUsingQueryClient will use queryClient to query whether the credentials in targetClient can perform
 // the actions listed in the statementEntries. queryClient will need iam:GetUser and iam:SimulatePrincipalPolicy
-func checkPermissionsUsingQueryClient(queryClient, targetClient *awsClient, policyDocument PolicyDocument,
+func checkPermissionsUsingQueryClient(client *awsClient, policyDocument PolicyDocument,
```

Comment:
We have both `queryClient` and `targetClient` here because we should actually be validating the ARN of the user created with cloudformation. I suggest leaving this and using the `targetClient` ARN when completing [OSD-4708](https://issues.redhat.com/browse/OSD-4708)

Reply:
mm that's ambiguous because still hasn't been implemented yet so we're creating structures below which are poor in terms of readability. Though, I'll take that suggestion and revert here as well

```go
hasPermissions, err := validatePolicyDocuments(c, c, policyDocuments, sParams)
```